packages/hdwallet-provider/src/LegacyConstructor.ts

```diff
@@ -0,0 +1,117 @@
+import {
+  MnemonicPhrase,
+  PrivateKey,
+  ProviderOrUrl,
+  AddressIndex,
+  NumberOfAddresses,
+  ShareNonce,
+  DerivationPath
+} from "./types";
+import { NewConstructor } from "./NewConstructor";
+
+/*
+ * namespace wrapper for old-style positional arguments
+ */
+export namespace LegacyConstructor {
```

Comment:
since we're making this a module of its own, we don't need the namespace - we can just have all these be top-level exports, and then do `import * as LegacyConstructor from "./LegacyConstructor"` on the import side.

Reply:
That makes a lot of sense. This will make me feel better about naming these files something like `constructor.ts` and `legacyConstructor.ts`.